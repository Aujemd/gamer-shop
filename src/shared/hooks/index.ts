export { useStoredCart } from "./use-stored-cart";
