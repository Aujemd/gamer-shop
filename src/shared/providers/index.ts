export * from "./data-fetching-provider";
