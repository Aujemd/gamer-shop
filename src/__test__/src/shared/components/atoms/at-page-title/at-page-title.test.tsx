import { render, screen } from "@testing-library/react";

import { AtPageTitle } from "@/shared";

describe("atPageTitle component", () => {
    it("should render correctly", () => {
        const { asFragment } = render(<AtPageTitle>Catalog</AtPageTitle>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders as an h1 element", () => {
        const { getByRole } = render(<AtPageTitle>Title</AtPageTitle>);
        const heading = getByRole("heading", { level: 1 });
        expect(heading).toBeInTheDocument();
    });

    it("applies custom className", () => {
        render(
            <AtPageTitle
                data-testid="at-page-title-test-id"
                className="custom-class"
            >
                Title
            </AtPageTitle>,
        );
        const heading = screen.getByTestId("at-page-title-test-id");
        expect(heading).toHaveClass("custom-class");
    });
});
