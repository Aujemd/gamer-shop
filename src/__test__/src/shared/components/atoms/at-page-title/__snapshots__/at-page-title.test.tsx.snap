// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atPageTitle component should render correctly 1`] = `
<DocumentFragment>
  <h1
    class="font-archivo font-bold text-2xl leading-7 tracking-sm text-center text-gray-medium lg:text-4xl lg:leading-10 "
  >
    Catalog
  </h1>
</DocumentFragment>
`;
